auslib/web/api_validator.py

```diff
@@ -0,0 +1,24 @@
+import logging
+from connexion.decorators.validation import ParameterValidator
+
+log = logging.getLogger(__name__)
+
+
+class BalrogParameterValidator(ParameterValidator):
+    """
+    This class is used to keep the flask "defaults" feature behavior on spec validation level.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(BalrogParameterValidator, self).__init__(*args, **kwargs)
+
+    def validate_path_parameter(self, param, request):
+        # Keep the base class behavior.
+        val = request.path_params.get(param['name'].replace('-', '_'))
+
+        # Supporting the flask add_url_rule "defaults" parameter to the swagger path parameter.
+        if val is None and 'default' in param:
+            val = param['default']
+            request.path_params[param['name']] = val
```

Comment:
I'm still a little confused about why this is necessary after a reread. According to the Swagger spec, defaults are supported...and Connexion also seems to support default parameters. I see that we get an error like this if we remove your validator:
```
<Response streamed [400 BAD REQUEST]>
{
  "detail": "Missing path parameter 'queryVersion'", 
  "status": 400, 
  "title": "Bad Request", 
  "type": "about:blank"
}

```

But I don't quite understanding why - can you explain that a bit?